### PR TITLE
allows specialWeapons to be defined by soldier type

### PR DESCRIPTION
**Pull request by memmaker**

### CODE REVIEW

File: src/Mod/RuleSoldier.cpp [hunk 1]
```diff
@@ -95,6 +95,7 @@ void RuleSoldier::load(const YAML::Node &node, Mod *mod, int listOrder, const Mo
 	}
 	_dogfightExperience.merge(node["dogfightExperience"].as<UnitStats>(_dogfightExperience));
 	_armor = node["armor"].as<std::string>(_armor);
+	_specWeapon = node["specialWeapon"].as<std::string>(_specWeapon);
```
Comment (Yankes) [inline src/Mod/RuleSoldier.cpp]: I think better is not use string but `const RuleItem*` and use `afterLoadHelper("research", this, _research, &RuleResearch::afterLoad);` function like that to link final object rule

**Comment by MeridianOXC** [comment]
So, why exactly is this needed? Why is special weapon on armor not enough?

**Comment by memmaker** [comment]
In my ruleset I had to use one specific armor per soldier type, just to have a special weapon for each soldier type. I have six soldier types currently implemented and will maybe add more. So that is six armor definitions in the ruleset which I would just not need if this was possible. And this is for one set of armors. Vanilla alone has several. Integrating with a mod that has own armors defined adds another interesting layer of copy/pasting just for having a special weapon for each soldier type. Yes, it is possible to do it via armor, but it is not what I would want to do.

**Comment by memmaker** [comment]
[ReportMod.zip](https://github.com/MeridianOXC/OpenXcom/files/4142021/ReportMod.zip)
This includes a ruleset which shows some of the possibilites.

**Comment by MeridianOXC** [comment]
No worries. I didn't say no. I just need to know the use case, to be able to judge whether it's a misuse or not.

**Comment by memmaker** [comment]
Well, for the most part if drastically reduces my ruleset and allows for the association with the special weapon where I would not need any armor definitions else.

File: src/Mod/RuleSoldier.cpp (hunk 1 above)
Comment (memmaker) [reply src/Mod/RuleSoldier.cpp]: I used the same way that this is done for the units and armors, or so I thought.
What does this do? `afterLoadHelper("research", this, _research, &RuleResearch::afterLoad);`

**Comment by MeridianOXC** [comment]
Merged.

File: src/Mod/RuleSoldier.cpp (hunk 1 above)
Comment (Yankes) [reply src/Mod/RuleSoldier.cpp]: This is only one case of more general functionality, you should look what exactly this function do. It link and check if all values read from yaml are correct and link correctly to other rule objects

**Pull request closed without merging**